src/Symfony/Component/EventDispatcher/EventListener.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\EventDispatcher;
+
+/**
+ * Helper class for single-event subscription.
+ *
+ * @author Yonel Ceruto <yonelceruto@gmail.com>
+ */
+abstract class EventListener implements EventSubscriberInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    final public static function getSubscribedEvents(): array
+    {
+        return [static::getEventName() => [static::getMethodName(), static::getPriority()]];
+    }
+
+    public static function getEventName(): string
+    {
+        $method = new \ReflectionMethod(static::class, static::getMethodName());
+
+        if (0 === $method->getNumberOfParameters()) {
+            throw new \LogicException(sprintf('Missing "%s" parameter in "%s::%s()" method.', Event::class, static::class, $method->getName()));
+        }
+
+        $parameter = $method->getParameters()[0];
+
+        if (null === $type = $parameter->getType()) {
+            throw new \LogicException(sprintf('Missing "%s" type-hint for "$%s" parameter in "%s::%s()" method.', Event::class, $parameter->getName(), static::class, $method->getName()));
+        }
+
+        if ($type->isBuiltin()) {
+            throw new \LogicException(sprintf('The "$%s" parameter defined in "%s::%s()" method must be a class or subclass of "%s", "%s" given.', $parameter->getName(), static::class, $method->getName(), Event::class, $type->getName()));
```

Comment:
I find the `class or subclass of Event` part confusing. A subclass of `Event` is a class. Maybe `…must be a class (e.g. a subclass of Event)…`?